android-components/components/feature/fxsuggest/src/main/java/mozilla/components/feature/fxsuggest/FxSuggestSuggestionProvider.kt

```diff
@@ -48,10 +48,11 @@ class FxSuggestSuggestionProvider(
             emptyList()
         } else {
             val providers = buildList() {
-                if (includeSponsoredSuggestions) {
+                val availableSuggestionTypes = FxSuggestNimbus.features.suggestionProvider.value().availableSuggestionTypes
+                if (includeSponsoredSuggestions && availableSuggestionTypes.contains(SuggestionType.AMP)) {
                     add(SuggestionProvider.AMP)
                 }
-                if (includeNonSponsoredSuggestions) {
+                if (includeNonSponsoredSuggestions && availableSuggestionTypes.contains(SuggestionType.WIKIPEDIA)) {
```

Comment:
Yeah, this definitely looks like the right call to me. Thanks for putting it together!